src/api/app/controllers/webui/project_controller.rb

```diff
@@ -606,52 +558,86 @@ def unlock
 
   private
 
-  def fill_status_cache
-    @repohash = {}
+  def monitor_buildresult
+    @legend = Buildresult::STATUS_DESCRIPTION
+
+    @name_filter = params[:pkgname]
+    @lastbuild_switch = params[:lastbuild]
+    if params[:defaults]
+      defaults = (begin
+                    Integer(params[:defaults])
+                  rescue ArgumentError
+                    1
+                  end) > 0
+    else
+      defaults = true
+    end
+    params['expansionerror'] = 1 if params['unresolvable']
+    monitor_set_filter(defaults)
+
+    find_opt = { project: @project, view: 'status', code: @status_filter,
+                 arch: @arch_filter, repository: @repo_filter }
+    find_opt[:lastbuild] = 1 if @lastbuild_switch.present?
+
+    buildresult = Buildresult.find_hashed(find_opt)
+    if buildresult.empty?
+      flash[:warning] = "No build results for project '#{@project}'"
+      redirect_to action: :show, project: params[:project]
+      return
```

Comment:
This is now only returning to the previous method and afterward `@buildresult_unavailable = true` is set although you are going to redirect... :thinking:

Reply:
Does it harm?

Reply:
no, but it is not very elegant... let's say it is a nitpick...